api: swagger: fix documentation for image push endpoint

thaJeztah opened the pull request:
- relates to https://github.com/moby/moby/issues/42053


The documentation was incorrect and didn't properly document the use of tags;

- Image push currently only accepts an image-name, not an ID / digest.
- When giving a name, it's expected to be without tag included; when including a tag, it is ignored.
- The tag parameter is required when pushing a single image (i.e., it does not default to "latest"); omitting the tag parameter will push all tags of the given image.


**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog.
It must be placed inside the below triple backticks section:
-->
```markdown changelog

```

**- A picture of a cute animal (not mandatory but encouraged)**

thaJeztah committed:
api: swagger: fix documentation for image push endpoint

The documentation was incorrect and didn't properly document the use of
tags;

- Image push currently only accepts an image-name, not an ID / digest.
- When giving a name, it's expected to be without tag included; when
  including a tag, it is ignored.
- The tag parameter is required when pushing a single image (i.e., it
  does not default to "latest"); omitting the tag parameter will push
  all tags of the given image.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
Opening as draft, because I'm not really sold on the wording 😂 - once we get to better wording, I can update the versioned copies that are used in the docs.

cc @dvdksn perhaps you have good suggestions

dvdksn commented on api/swagger.yaml:
```diff
@@ -9231,12 +9231,24 @@ paths:
           in: "path"
-          description: "Image name or ID."
+          description: |
+            Name of the image to push, for example, `registry.example.com/myimage`.
```
```suggestion
            Name of the image to push. For example, `registry.example.com/myimage`.
```

dvdksn commented on api/swagger.yaml:
```diff
@@ -9231,12 +9231,24 @@ paths:
-          description: "Image name or ID."
+          description: |
+            Name of the image to push, for example, `registry.example.com/myimage`.
+            The image must be present in the local image store and tagged with
+            the given name.
```
maybe avoid "tagged" with here? 😅 

```suggestion
            The image must be present in the local image store with the same name.
```

dvdksn commented on api/swagger.yaml:
```diff
@@ -9231,12 +9231,24 @@ paths:
+            the given name.
+
+            The name should be provided without tag; if a tag is provided, it
+            is ignored (for example, `registry.example.com/myimage:latest` is
+            considered equivalent to ``registry.example.com/myimage`).
```
```suggestion
            is ignored. For example, `registry.example.com/myimage:latest` is
            equivalent to `registry.example.com/myimage`.
```

thaJeztah committed:
docs: api: fix documentation for image push endpoint (API v1.25-v1.47)

The documentation was incorrect and didn't properly document the use of
tags;

- Image push currently only accepts an image-name, not an ID / digest.
- When giving a name, it's expected to be without tag included; when
  including a tag, it is ignored.
- The tag parameter is required when pushing a single image (i.e., it
  does not default to "latest"); omitting the tag parameter will push
  all tags of the given image.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
Updated; also updated the versioned files used in the docs

dvdksn reviewed (approved):
LGTM
